_posts/2020-09-18-ai-ethics-optimization-problem/ai-ethics-optimization-problem.Rmd

```diff
@@ -0,0 +1,172 @@
+
+[^1]: For brevity, I'll be subsuming deep learning and other contemporary machine learning methods under *AI*, following common(-ish) usage.
+
+To eliminate one possible misunderstanding right from the start: When I'm talking about "faster and faster" progress, I'm not over-hyping things. I'm far from thinking that *AI* is close to "solving" problems like language understanding, concept learning and their likes -- the kind of problems some would argue hybrid models were needed for [@2020arXiv200206177M]. The thing is that it doesn't matter. It is exactly the kinds of things *AI* *does* do so well that lend themselves to misuse. It is people we should be afraid of, not machines[^2].
+
+[^2]: I can't hope to express this better than Maciej Cegłowski did [here](https://idlewords.com/talks/superintelligence.htm), so I won't elaborate on that topic any further.
+
+Back to the why. Over time, it increasingly appeared to me that writing regularly about *AI* in a technical way, but *not ever* writing about its misuse, was ethically questionable in itself. However, I also became increasingly conscious of the fact that with a topic like this, once you enter the political realm -- and that we *have to* is the main point of this text --, likelihood rises that people will disagree, or worse, feel offended for reasons not anticipated by the writer. Some will find this too radical, some not radical (*explicit*) enough.
+
+But when the alternative is to stay silent, it seems better to try and do one's best.
+
+Let's start with two terms whose recent rise in popularity matches that of *AI* as a whole: bias and fairness.
+
+## But we *are* doing a lot to improve fairness and remove bias, aren't we?
+
+Search for "algorithmic fairness", "deep learning fairness" or something similar, and you'll find lots of papers, tools, guidelines ... more than you have time to read. And bias: Haven't we all heard about image recognition models failing on black people, women, and black women in particular; about machine translation incorporating gender stereotypes; about search engine results reflecting racism and discrimination? Given enough media attention, the respective algorithms get "fixed"; what's more, we may also safely assume that overall, researchers developing new models will probe for these exact kinds of failures. So as a community, we *do* care about bias, don't we?
+
+Re: fairness. It's true that there are many attempts to increase algorithmic fairness. But as Ben Green [@AlgorithmicRealism], who I'll cite a lot in this text, points out, most of this work does so via rigid formalization, as if fairness -- its [conflicting definitions](https://www.youtube.com/watch?v=jIXIuYdnyyk) notwithstanding -- were an objective quantity, a metric that could be optimized just like the metrics we usually optimize in machine learning.
```

Comment:
Interesting!

I think I would probably also approach this topic by way of an analogy:

- In the human world we have lawmakers working on legal frameworks. If the lawmakers are ethical and also competent at what they do, they would make laws that's at least penalize some unethical behaviors. Other unethical behaviors might slip through the crack though.
- I would say in the computing world AI researchers have a similar "lawyering" responsibility. The ethical ones would work on frameworks to help AI avoid some unethical choices, as those choices would be penalized algorithmically (where optimization plays a role), but still, there is no guarantee any such framework can rule out all unethical choices. So we cannot afford to think there is some kind of silver bullet and become complacent.

Anyhow, just my $0.02 on this topic 🙂

Reply:
Hey Yitao :-)

... I'd say that overall I agree, just IMO this has to be put a lot more strongly.
Like you hint to ("_if_  lawmakers are ..."), legislation as well takes place in a context of societal norms (not to mention _lobbying_). 

I don't know enough about law, but at least I'd hope that practitioners be at least in some way prepared for thinking about societal context - which AI researchers (computer scientists in general) are not. But if we assume they were (for example, because ethics had been incorporated into the CS curriculum), there still is the problem that they don't do their work in a void. They are employed (and extremely well paid!) by companies whose business incentives are misaligned with respecting rights like privacy, autonomy ... And whatever they develop is deployed by someone - companies, governments - that have a clear interest not to respect these rights either.

--- Re. constrained optimization (didn't see your edit at first). Here the point is that you can't compare, say, rights with accuracy (you can't quantize some things). Even if you could, who defines the constraints?